products/apigateway/api.yaml

```diff
@@ -0,0 +1,227 @@
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: |
+          A user-visible name for the API.
+      - !ruby/object:Api::Type::String
+        name: 'managedService'
+        input: true
+        description: |
+          Immutable. The name of a Google Managed Service ( https://cloud.google.com/service-infrastructure/docs/glossary#managed).
+          If not specified, a new Service will automatically be created in the same project as this API.
+      - !ruby/object:Api::Type::Time
+        name: 'createTime'
+        description: Creation timestamp in RFC3339 text format.
+        output: true
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: |
+          Resource labels to represent user-provided metadata.
+  - !ruby/object:Api::Resource
+    name: 'ApiConfig'
+    min_version: beta
+    create_url: projects/{{project}}/locations/global/apis/{{api}}/configs?apiConfigId={{api_config_id}}
+    self_link: projects/{{project}}/locations/global/apis/{{api}}/configs/{{api_config_id}}
+    base_url: projects/{{project}}/locations/global/apis/{{api}}/configs
+    read_query_params: '?view=FULL'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A consumable API that can be used by multiple Gateways.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/api-gateway/docs/quickstart'
+      api: 'https://cloud.google.com/api-gateway/docs/reference/rest/v1beta/projects.locations.apis.configs'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: api
+        description: |
+          The API to attach the config to.
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: apiConfigId
+        description: |
+          Identifier to assign to the API Config. Must be unique within scope of the parent resource.
+        required: true
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: The resource name of the API Config.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: |
+          A user-visible name for the API.
+      - !ruby/object:Api::Type::String
+        name: 'serviceConfigId'
+        output: true
+        description: |
+          The ID of the associated Service Config (https://cloud.google.com/service-infrastructure/docs/glossary#config).
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: |
+          Resource labels to represent user-provided metadata.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'gatewayConfig'
+        description: |
+          Immutable. Gateway specific configuration.
```

Comment:
If this is immutable should it be marked `input: true`?

Reply:
yeah